### PR TITLE
feat: Add fill_diagonal_tensor_ to Paddle Frontend

### LINKED ISSUE #26892: fill_diagonal_tensor_

### PR DESCRIPTION
**Pull request by mahmoud99-cell**
# PR Description
I tested the method in a separate notebook by creating a custom class with this function only and using a normal tensor object from paddle package, and it worked correctly, of course system testing is the one who will tell if it is integrated correctly. 

Still need to test inplace behavior and update it (use ivy.update_inplace(self, y)). 

I created the tests but failed to run due to windows fatal threading error. Also pre-commit fails when I ran it, followed [the instructions](https://unify.ai/docs/ivy/overview/contributing/setting_up.html#pre-commit).

Command:
`pytest ivy\ivy_tests\test_ivy\test_frontends\test_paddle\test_tensor\test_tensor.py::fill_diagonal_tensor_ --no-header --no-summary --disable-pytest-warnings --backend all --hypothesis-verbosity verbose`

❌ Error  :
  test session starts
collecting ... Windows fatal exception: access violation
Thread 0x00004714 (most recent call first):
  File "D:\Anaconda3\envs\ivy_dev\lib\threading.py", line 324 in wait
  File "D:\Anaconda3\envs\ivy_dev\lib\threading.py", line 600 in wait
  File "D:\Anaconda3\envs\ivy_dev\lib\threading.py", line 1356 in run
  File "D:\Anaconda3\envs\ivy_dev\lib\threading.py", line 1009 in _bootstrap_inner
  File "D:\Anaconda3\envs\ivy_dev\lib\threading.py", line 966 in _bootstrap

## Related Issue

Closes #26892 

## Checklist

- [x] Did you add a function? yes
- [x] Did you add the tests? yes
- [ ] Did you run your tests and are your tests passing? no
- [ ] Did pre-commit not fail on any check? yes, package not running ImportError: DLL load failed while importing _ctypes:
- [x] Did you follow the steps we provided? yes

### Socials
linkedin: https://www.linkedin.com/in/mahmoud-nabil-9a75b11a5/

### CODE REVIEW

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 1]
```diff
@@ -908,3 +908,51 @@ def gather(self, y, name=None):
...
+            return x
+        else:
+            # Create a copy of 'x' and apply the modification to the copy
+            result = x.clone()
+            result[:diaglen, diag_start:diag_end] = y
+            return result
+
+    # Paddle source code implementation
+    def fill_diagonal_tensor_(self, y, offset=0, dim1=0, dim2=1, name=None):
+        self = self.fill_diagonal_tensor_impl(
```
Comment (KareemMAX) [inline ivy/functional/frontends/paddle/tensor/tensor.py:955]: I'm not really sure if assigning `self` is valid (or a good practice) in python, you are basically assigning memory address by itself.

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 2]
```diff
@@ -908,3 +908,51 @@ def gather(self, y, name=None):
...
+        print(inshape)
+        assert dim1 < len(inshape) and dim1 >= -len(
+            inshape
+        ), "dim1 should be between [-rank, rank) in fill_diagonal_tensor_"
+        assert dim2 < len(inshape) and dim2 >= -len(
+            inshape
+        ), "dim2 should be between [-rank, rank) in fill_diagonal_tensor_"
+        assert len(inshape) >= 2, "Tensor dims should be >= 2 in fill_diagonal_tensor_"
+        dim1 %= len(inshape)
+        dim2 %= len(inshape)
```
Comment (KareemMAX) [inline ivy/functional/frontends/paddle/tensor/tensor.py:923]: Are you sure that is computationally correct? Why don't you use more simpler terms?

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py [hunk 3]
```diff
@@ -330,6 +330,46 @@ def test_paddle__reshape(
     )
 
 
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="paddle.to_tensor",
+    method_name="fill_diagonal_tensor_",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
```
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:338]: Can you use `valid` instead?

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:1]: You seem to be missing the other arguments of the function

**Review by KareemMAX — CHANGES_REQUESTED** [review]
Hey @mahmoud99-cell,
I've pointed out some comments that I would like you to resolve. Overall, I think you might be over complicating the function a bit as it's extremely hard to understand (therefore maintain). Have you tried using masking instead?

Also there are some irrelevant changes that you have made, please revert them back

**Comment by KareemMAX** [comment]
> I tested the method in a separate notebook by creating a custom class with this function only and using a normal tensor object from paddle package, and it worked correctly, of course system testing is the one who will tell if it is integrated correctly.

Seems like those error are Windows specific, have you tried another system? Maybe docker (as it's the most predictable)

**Comment by mahmoud99-cell** [comment]
> > I tested the method in a separate notebook by creating a custom class with this function only and using a normal tensor object from paddle package, and it worked correctly, of course system testing is the one who will tell if it is integrated correctly.
> 
> Seems like those error are Windows specific, have you tried another system? Maybe docker (as it's the most predictable)

Can you test the pytest function on your end? Can you send me your discord ID ?

**Review by KareemMAX — CHANGES_REQUESTED** [review]
<details><summary>Error logs</summary>

```
Use Database in ReadOnly Mode with local caching !
============================= test session starts ==============================
platform linux -- Python 3.10.0, pytest-7.4.2, pluggy-1.3.0 -- /opt/miniconda/envs/multienv/bin/python
cachedir: .pytest_cache
hypothesis profile 'ivy_profile' -> max_examples=25, print_blob=True, deadline=timedelta(milliseconds=500000), suppress_health_check=[HealthCheck.too_slow, HealthCheck.filter_too_much, HealthCheck.data_too_large], database=MultiplexedDatabase(DirectoryBasedExampleDatabase('/workspaces/ivy/.hypothesis/examples/'), ReadOnlyDatabase(RedisExampleDatabase(Redis<ConnectionPool<Connection<host=redis-17011.c259.us-central1-2.gce.cloud.redislabs.com,port=17011,db=0>>>, expire_after=datetime.timedelta(days=8))))
metadata: {'Python': '3.10.0', 'Platform': 'Linux-6.2.0-1014-azure-x86_64-with-glibc2.28', 'Packages': {'pytest': '7.4.2', 'pluggy': '1.3.0'}, 'Plugins': {'anyio': '4.0.0', 'hypothesis': '6.87.0', 'metadata': '3.0.0', 'json-report': '1.5.0'}}
backend(s): all
device: cpu
number of Hypothesis examples: 25
rootdir: /workspaces/ivy
plugins: anyio-4.0.0, hypothesis-6.87.0, metadata-3.0.0, json-report-1.5.0
collecting ... collected 5 items

ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py::test_paddle_fill_diagonal_tensor_[cpu-numpy-False-False] FAILED [ 20%]
ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py::test_paddle_fill_diagonal_tensor_[cpu-jax-False-False] FAILED [ 40%]
ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py::test_paddle_fill_diagonal_tensor_[cpu-tensorflow-False-False] FAILED [ 60%]
ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py::test_paddle_fill_diagonal_tensor_[cpu-torch-False-False] FAILED [ 80%]
ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py::test_paddle_fill_diagonal_tensor_[cpu-paddle-False-False] FAILED [100%]

=================================== FAILURES ===================================
___________ test_paddle_fill_diagonal_tensor_[cpu-numpy-False-False] ___________

    @handle_frontend_method(
>       class_tree=CLASS_TREE,
        init_tree="paddle.to_tensor",
        method_name="fill_diagonal_tensor_",
        dtype_and_x=helpers.dtype_and_values(
            available_dtypes=helpers.get_dtypes("valid"),
            num_arrays=2,
            shared_dtype=True,
        ),
        test_inplace=st.just(True),
    )

ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:334: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:355: in test_paddle_fill_diagonal_tensor_
    helpers.test_frontend_method(
ivy_tests/test_ivy/helpers/function_testing.py:2052: in test_frontend_method
    frontend_ret_ins = copy_ins.__getattribute__(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = ivy.frontends.paddle.Tensor(shape=[], dtype=bool, True)
y = ivy.array(False), offset = 0, dim1 = 0, dim2 = 1, name = None

    def fill_diagonal_tensor_(self, y, offset=0, dim1=0, dim2=1, name=None):
>       self._ivy_array = self.fill_diagonal_tensor_impl(
            self, y, offset=offset, dim1=dim1, dim2=dim2, inplace=True
        )
E       AttributeError: 'Tensor' object has no attribute 'fill_diagonal_tensor_impl'. Did you mean: '_fill_diagonal_tensor_impl'?
E       Falsifying example: test_paddle_fill_diagonal_tensor_(
E           frontend='paddle',
E           on_device='cpu',
E           backend_fw='numpy',
E           dtype_and_x=(['bool', 'bool'], [array(True), array(False)]),
E           init_flags=FrontendInitTestFlags(
E               num_positional_args=1,
E               as_variable=[False],
E               native_arrays=[False],
E           ),
E           frontend_method_data=FrontendMethodData(ivy_init_module='ivy.functional.frontends.paddle', framework_init_module='paddle', init_name='to_tensor', method_name='fill_diagonal_tensor_'),
E           method_flags=FrontendMethodTestFlags(
E               num_positional_args=0,
E               as_variable=[False],
E               native_arrays=[False],
E               precision_mode=False,
E               inplace=True,
E               test_trace=False,
E               generate_frontend_arrays=False,
E           ),
E       )
E       
E       You can reproduce this example by temporarily adding @reproduce_failure('6.87.0', b'AAgAAQAAAAAAAAAAAA==') as a decorator on your test case

ivy/functional/frontends/paddle/tensor/tensor.py:928: AttributeError
____________ test_paddle_fill_diagonal_tensor_[cpu-jax-False-False] ____________

    @handle_frontend_method(
>       class_tree=CLASS_TREE,
        init_tree="paddle.to_tensor",
        method_name="fill_diagonal_tensor_",
        dtype_and_x=helpers.dtype_and_values(
            available_dtypes=helpers.get_dtypes("valid"),
            num_arrays=2,
            shared_dtype=True,
        ),
        test_inplace=st.just(True),
    )

ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:334: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:355: in test_paddle_fill_diagonal_tensor_
    helpers.test_frontend_method(
ivy_tests/test_ivy/helpers/function_testing.py:2052: in test_frontend_method
    frontend_ret_ins = copy_ins.__getattribute__(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = ivy.frontends.paddle.Tensor(shape=[], dtype=bool, False)
y = ivy.array(False), offset = 0, dim1 = 0, dim2 = 1, name = None

    def fill_diagonal_tensor_(self, y, offset=0, dim1=0, dim2=1, name=None):
>       self._ivy_array = self.fill_diagonal_tensor_impl(
            self, y, offset=offset, dim1=dim1, dim2=dim2, inplace=True
        )
E       AttributeError: 'Tensor' object has no attribute 'fill_diagonal_tensor_impl'. Did you mean: '_fill_diagonal_tensor_impl'?
E       Falsifying example: test_paddle_fill_diagonal_tensor_(
E           # The test always failed when commented parts were varied together.
E           frontend='paddle',
E           on_device='cpu',
E           backend_fw='jax',
E           dtype_and_x=(['bool', 'bool'],
E            [array(False), array(False)]),  # or any other generated value
E           init_flags=FrontendInitTestFlags(
E               num_positional_args=1,
E               as_variable=[False],
E               native_arrays=[False],
E           ),  # or any other generated value
E           frontend_method_data=FrontendMethodData(ivy_init_module='ivy.functional.frontends.paddle', framework_init_module='paddle', init_name='to_tensor', method_name='fill_diagonal_tensor_'),
E           method_flags=FrontendMethodTestFlags(
E               num_positional_args=0,
E               as_variable=[False],
E               native_arrays=[False],
E               precision_mode=False,
E               inplace=True,
E               test_trace=False,
E               generate_frontend_arrays=False,
E           ),  # or any other generated value
E       )
E       
E       You can reproduce this example by temporarily adding @reproduce_failure('6.87.0', b'AXic42RABgAAjAAK') as a decorator on your test case

ivy/functional/frontends/paddle/tensor/tensor.py:928: AttributeError
________ test_paddle_fill_diagonal_tensor_[cpu-tensorflow-False-False] _________

    @handle_frontend_method(
>       class_tree=CLASS_TREE,
        init_tree="paddle.to_tensor",
        method_name="fill_diagonal_tensor_",
        dtype_and_x=helpers.dtype_and_values(
            available_dtypes=helpers.get_dtypes("valid"),
            num_arrays=2,
            shared_dtype=True,
        ),
        test_inplace=st.just(True),
    )

ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:334: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:355: in test_paddle_fill_diagonal_tensor_
    helpers.test_frontend_method(
ivy_tests/test_ivy/helpers/function_testing.py:2052: in test_frontend_method
    frontend_ret_ins = copy_ins.__getattribute__(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = ivy.frontends.paddle.Tensor(shape=[], dtype=bool, False)
y = ivy.array(False), offset = 0, dim1 = 0, dim2 = 1, name = None

    def fill_diagonal_tensor_(self, y, offset=0, dim1=0, dim2=1, name=None):
>       self._ivy_array = self.fill_diagonal_tensor_impl(
            self, y, offset=offset, dim1=dim1, dim2=dim2, inplace=True
        )
E       AttributeError: 'Tensor' object has no attribute 'fill_diagonal_tensor_impl'. Did you mean: '_fill_diagonal_tensor_impl'?
E       Falsifying example: test_paddle_fill_diagonal_tensor_(
E           # The test always failed when commented parts were varied together.
E           frontend='paddle',
E           on_device='cpu',
E           backend_fw='tensorflow',
E           dtype_and_x=(['bool', 'bool'],
E            [array(False), array(False)]),  # or any other generated value
E           init_flags=FrontendInitTestFlags(
E               num_positional_args=1,
E               as_variable=[False],
E               native_arrays=[False],
E           ),  # or any other generated value
E           frontend_method_data=FrontendMethodData(ivy_init_module='ivy.functional.frontends.paddle', framework_init_module='paddle', init_name='to_tensor', method_name='fill_diagonal_tensor_'),
E           method_flags=FrontendMethodTestFlags(
E               num_positional_args=0,
E               as_variable=[False],
E               native_arrays=[False],
E               precision_mode=False,
E               inplace=True,
E               test_trace=False,
E               generate_frontend_arrays=False,
E           ),  # or any other generated value
E       )
E       
E       You can reproduce this example by temporarily adding @reproduce_failure('6.87.0', b'AXic42RABgAAjAAK') as a decorator on your test case

ivy/functional/frontends/paddle/tensor/tensor.py:928: AttributeError
___________ test_paddle_fill_diagonal_tensor_[cpu-torch-False-False] ___________

    @handle_frontend_method(
>       class_tree=CLASS_TREE,
        init_tree="paddle.to_tensor",
        method_name="fill_diagonal_tensor_",
        dtype_and_x=helpers.dtype_and_values(
            available_dtypes=helpers.get_dtypes("valid"),
            num_arrays=2,
            shared_dtype=True,
        ),
        test_inplace=st.just(True),
    )

ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:334: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:355: in test_paddle_fill_diagonal_tensor_
    helpers.test_frontend_method(
ivy_tests/test_ivy/helpers/function_testing.py:2052: in test_frontend_method
    frontend_ret_ins = copy_ins.__getattribute__(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = ivy.frontends.paddle.Tensor(shape=[], dtype=bool, False)
y = ivy.array(False), offset = 0, dim1 = 0, dim2 = 1, name = None

    def fill_diagonal_tensor_(self, y, offset=0, dim1=0, dim2=1, name=None):
>       self._ivy_array = self.fill_diagonal_tensor_impl(
            self, y, offset=offset, dim1=dim1, dim2=dim2, inplace=True
        )
E       AttributeError: 'Tensor' object has no attribute 'fill_diagonal_tensor_impl'. Did you mean: '_fill_diagonal_tensor_impl'?
E       Falsifying example: test_paddle_fill_diagonal_tensor_(
E           # The test always failed when commented parts were varied together.
E           frontend='paddle',
E           on_device='cpu',
E           backend_fw='torch',
E           dtype_and_x=(['bool', 'bool'],
E            [array(False), array(False)]),  # or any other generated value
E           init_flags=FrontendInitTestFlags(
E               num_positional_args=1,
E               as_variable=[False],
E               native_arrays=[False],
E           ),  # or any other generated value
E           frontend_method_data=FrontendMethodData(ivy_init_module='ivy.functional.frontends.paddle', framework_init_module='paddle', init_name='to_tensor', method_name='fill_diagonal_tensor_'),
E           method_flags=FrontendMethodTestFlags(
E               num_positional_args=0,
E               as_variable=[False],
E               native_arrays=[False],
E               precision_mode=False,
E               inplace=True,
E               test_trace=False,
E               generate_frontend_arrays=False,
E           ),  # or any other generated value
E       )
E       
E       You can reproduce this example by temporarily adding @reproduce_failure('6.87.0', b'AXic42RABgAAjAAK') as a decorator on your test case

ivy/functional/frontends/paddle/tensor/tensor.py:928: AttributeError
__________ test_paddle_fill_diagonal_tensor_[cpu-paddle-False-False] ___________

    @handle_frontend_method(
>       class_tree=CLASS_TREE,
        init_tree="paddle.to_tensor",
        method_name="fill_diagonal_tensor_",
        dtype_and_x=helpers.dtype_and_values(
            available_dtypes=helpers.get_dtypes("valid"),
            num_arrays=2,
            shared_dtype=True,
        ),
        test_inplace=st.just(True),
    )

ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:334: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:355: in test_paddle_fill_diagonal_tensor_
    helpers.test_frontend_method(
ivy_tests/test_ivy/helpers/function_testing.py:2052: in test_frontend_method
    frontend_ret_ins = copy_ins.__getattribute__(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = ivy.frontends.paddle.Tensor(shape=[], dtype=bool, False)
y = ivy.array(False), offset = 0, dim1 = 0, dim2 = 1, name = None

    def fill_diagonal_tensor_(self, y, offset=0, dim1=0, dim2=1, name=None):
>       self._ivy_array = self.fill_diagonal_tensor_impl(
            self, y, offset=offset, dim1=dim1, dim2=dim2, inplace=True
        )
E       AttributeError: 'Tensor' object has no attribute 'fill_diagonal_tensor_impl'. Did you mean: '_fill_diagonal_tensor_impl'?
E       Falsifying example: test_paddle_fill_diagonal_tensor_(
E           # The test always failed when commented parts were varied together.
E           frontend='paddle',
E           on_device='cpu',
E           backend_fw='paddle',
E           dtype_and_x=(['bool', 'bool'],
E            [array(False), array(False)]),  # or any other generated value
E           init_flags=FrontendInitTestFlags(
E               num_positional_args=1,
E               as_variable=[False],
E               native_arrays=[False],
E           ),  # or any other generated value
E           frontend_method_data=FrontendMethodData(ivy_init_module='ivy.functional.frontends.paddle', framework_init_module='paddle', init_name='to_tensor', method_name='fill_diagonal_tensor_'),
E           method_flags=FrontendMethodTestFlags(
E               num_positional_args=0,
E               as_variable=[False],
E               native_arrays=[False],
E               precision_mode=False,
E               inplace=True,
E               test_trace=False,
E               generate_frontend_arrays=False,
E           ),  # or any other generated value
E       )
E       
E       You can reproduce this example by temporarily adding @reproduce_failure('6.87.0', b'AXic42RABgAAjAAK') as a decorator on your test case

ivy/functional/frontends/paddle/tensor/tensor.py:928: AttributeError
=============================== warnings summary ===============================
../../opt/fw/tensorflow/_distutils_hack/__init__.py:18
  /opt/fw/tensorflow/_distutils_hack/__init__.py:18: UserWarning: Distutils was imported before Setuptools, but importing Setuptools also replaces the `distutils` module in `sys.modules`. This may lead to undesirable behaviors or errors. To avoid these issues, avoid using distutils directly, ensure that setuptools is installed in the traditional way (e.g. not an editable install), and/or make sure that setuptools is always imported before distutils.
    warnings.warn(

../../opt/fw/tensorflow/_distutils_hack/__init__.py:33
  /opt/fw/tensorflow/_distutils_hack/__init__.py:33: UserWarning: Setuptools is replacing distutils.
    warnings.warn("Setuptools is replacing distutils.")

../../opt/fw/tensorflow/setuptools/sandbox.py:13
  /opt/fw/tensorflow/setuptools/sandbox.py:13: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    import pkg_resources

../../opt/fw/tensorflow/pkg_resources/__init__.py:2871
  /opt/fw/tensorflow/pkg_resources/__init__.py:2871: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('sphinxcontrib')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)

ivy/utils/exceptions.py:386
  /workspaces/ivy/ivy/utils/exceptions.py:386: UserWarning: The current backend: 'jax' does not support inplace updates natively. Ivy would quietly create new arrays when using inplace updates with this backend, leading to memory overhead (same applies for views). If you want to control your memory management, consider doing ivy.set_inplace_mode('strict') which should raise an error whenever an inplace update is attempted with this backend.
    warnings.warn(

ivy/utils/exceptions.py:386
  /workspaces/ivy/ivy/utils/exceptions.py:386: UserWarning: The current backend: 'tensorflow' does not support inplace updates natively. Ivy would quietly create new arrays when using inplace updates with this backend, leading to memory overhead (same applies for views). If you want to control your memory management, consider doing ivy.set_inplace_mode('strict') which should raise an error whenever an inplace update is attempted with this backend.
    warnings.warn(

ivy/utils/exceptions.py:386
  /workspaces/ivy/ivy/utils/exceptions.py:386: UserWarning: The current backend: 'paddle' does not support inplace updates natively. Ivy would quietly create new arrays when using inplace updates with this backend, leading to memory overhead (same applies for views). If you want to control your memory management, consider doing ivy.set_inplace_mode('strict') which should raise an error whenever an inplace update is attempted with this backend.
    warnings.warn(

ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py: 17 warnings
  /workspaces/ivy/ivy/__init__.py:1000: RuntimeWarning: overflow encountered in power
    mags = 10 ** (sig_fig - 1 - np.floor(np.log10(x_positive)))

ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py: 17 warnings
  /workspaces/ivy/ivy/__init__.py:1001: RuntimeWarning: invalid value encountered in divide
    return np.round(x * mags) / mags

ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py: 1753 warnings
  /opt/fw/tensorflow/tensorflow/python/framework/ops.py:1035: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
    return int(self._numpy())

ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py: 15 warnings
  /workspaces/ivy/ivy/__init__.py:1000: RuntimeWarning: overflow encountered in scalar power
    mags = 10 ** (sig_fig - 1 - np.floor(np.log10(x_positive)))

ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py: 15 warnings
  /workspaces/ivy/ivy/__init__.py:1001: RuntimeWarning: invalid value encountered in scalar divide
    return np.round(x * mags) / mags

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
[32m=============================== 0.0% of 5 passed ========================================================== short test summary info ============================
FAILED ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py::test_paddle_fill_diagonal_tensor_[cpu-numpy-False-False]
FAILED ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py::test_paddle_fill_diagonal_tensor_[cpu-jax-False-False]
FAILED ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py::test_paddle_fill_diagonal_tensor_[cpu-tensorflow-False-False]
FAILED ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py::test_paddle_fill_diagonal_tensor_[cpu-torch-False-False]
FAILED ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py::test_paddle_fill_diagonal_tensor_[cpu-paddle-False-False]
================= 5 failed, 1824 warnings in 690.17s (0:11:30) =================
```

</details>

As shown above your code is failing the tests. My discord id is `kareemmax` although I'd rather if you posted your question on the public channels (for other contributors to see)

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py [hunk 4]
```diff
@@ -4542,6 +4582,43 @@ def test_paddle_tanh_(
...
+            "y": x[1],
+        },
+        frontend_method_data=frontend_method_data,
+        init_flags=init_flags,
+        method_flags=method_flags,
+        frontend=frontend,
+        on_device=on_device,
+    )
+
+
```
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:4621]: Those changes are irrelevant

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 5]
```diff
@@ -925,3 +925,47 @@ def gather(self, y, name=None):
...
+            )[:min_size, :min_size]
+
+        x._ivy_array = x._ivy_array.multiply(1 - dim1_mask).multiply(
+            1 - dim2_mask
+        ) + y._ivy_array.multiply(dim1_mask).multiply(dim2_mask)
+
+        return x._ivy_array
+
+    def fill_diagonal_tensor_(self, y, offset=0, dim1=0, dim2=1, name=None):
+        self._ivy_array = self._fill_diagonal_tensor_impl(
```
Comment (KareemMAX) [inline ivy/functional/frontends/paddle/tensor/tensor.py:968]: I'm not sure if that still is the right way of implementing the function

**Review by KareemMAX — CHANGES_REQUESTED** [review]

**Comment by Ishticode** [comment]
Thank you very much for the PR. Closing this due to lack of activity with changes requested remaining unresolved for over 6 months. Please feel free to reopen if you would like to continue working on this. :)

**Pull request closed without merging**